### PR TITLE
ttml to vtt region settings migration

### PR DESCRIPTION
**Pull request by discovery-avishekgulshan**
1. WEBVTT:  Nil pointer exception when the region doesn't have parent style.
2. TTML to VTT region settings migrated.
3. WebVTT read cue settings from parent style if no setting present inline
4. TTML to VTT cue position settings migrated.

### CODE REVIEW

**Comment by discovery-avishekgulshan** [comment]
Hi, can you please review this PR once?

File: subtitles.go [hunk 1]
```diff
@@ -266,27 +266,40 @@ func (sa *StyleAttributes) propagateTeletextAttributes() {
...
 	if sa.TTMLTextAlign != nil {
 		sa.WebVTTAlign = *sa.TTMLTextAlign
 	}
-}
-
-func (sa *StyleAttributes) propagateTTMLRegionAttributes() {
 	if sa.TTMLExtent != nil {
+		//region settings
 		lineHeight := 5 //assuming height of line as 5.33vh
 		dimensions := strings.Split(*sa.TTMLExtent, " ")
```
Comment (asticode) [inline subtitles.go:277]: Can you wrap below logic inside `if len(dimensions) > 1`

Indeed if `len(dimensions) == 1`, using `dimensions[1]` will panic

File: subtitles.go [hunk 2]
```diff
@@ -266,27 +266,40 @@ func (sa *StyleAttributes) propagateTeletextAttributes() {
...
+			sa.WebVTTSize = dimensions[0]
+		}
 	}
 	if sa.TTMLOrigin != nil {
+		//region settings
 		sa.WebVTTRegionAnchor = "0%,0%"
 		sa.WebVTTViewportAnchor = strings.ReplaceAll(strings.TrimSpace(*sa.TTMLOrigin), " ", ",")
+		sa.WebVTTScroll = "up"
+		//cue settings
+		coordinates := strings.Split(*sa.TTMLOrigin, " ")
```
Comment (asticode) [inline subtitles.go:295]: Can you wrap below logic inside `if len(coordinates) > 1`

Indeed if `len(coordinates) == 1`, using `coordinates[1]` will panic

**Review by asticode — CHANGES_REQUESTED** [review]
Thanks for the PR.

Only a few `len()` checks missing and we're good to go.

Ping me once you've fixed the issues.

File: subtitles.go (hunk 2 above)
Comment (discovery-avishekgulshan) [reply subtitles.go:295]: Thanks, for the review. Done

File: subtitles.go (hunk 1 above)
Comment (discovery-avishekgulshan) [reply subtitles.go:277]: Done

**Review by asticode — CHANGES_REQUESTED** [review]
Tests are now failing, can you fix them?

**Comment by discovery-avishekgulshan** [comment]
> Tests are now failing, can you fix them?

Fixed.

**Comment by asticode** [comment]
FYI I've created a `v0.14.0` tag

**Comment by discovery-avishekgulshan** [comment]
> FYI I've created a `v0.14.0` tag

Thank you. Can we connect on style migration from TTML to VTT?

**Comment by asticode** [comment]
What do you mean ?

**Comment by discovery-avishekgulshan** [comment]
> What do you mean ?

Nothing ignore that.

I will be adding Style migration from TTML to VTT as well and will require some changes with how TTML is being parsed. Once i raise the PR we can discuss it.

**Pull request merged**